self-run-agents/instances/main.tf

```diff
@@ -6,16 +6,13 @@ provider "aws" {
 module "x64-large-build-pool" {
   source = "./azp-build-asg"
 
-  ami_prefix                 = "envoy-azp-x64-large"
-  aws_account_id             = "457956385456"
-  azp_pool_name              = "x64-large"
-  azp_token                  = var.azp_token
-  disk_size_gb               = 2000
-  guaranteed_instances_count = 2
-  instance_type              = "r5.4xlarge"
-
-  sns_lifecycle_arn     = aws_sns_topic.lifecycle_updates.arn
-  sns_lifecyle_role_arn = aws_iam_role.asg_sns_role.arn
+  ami_prefix           = "envoy-azp-x64-large"
+  aws_account_id       = "457956385456"
+  azp_pool_name        = "x64-large"
+  azp_token            = var.azp_token
+  disk_size_gb         = 2000
+  idle_instances_count = 1
+  instance_type        = "r5.8xlarge"
```

Comment:
uhhh didn't notice this first time through, why are we switching to bigger instances here (for both of these pools)? I didn't think we need bigger instances which cost so much more, even with spot pricing?

Reply:
This is in exchange for https://github.com/envoyproxy/envoy/pull/11633, cache costed us significantly more due to the nature of huge binary in coverage runs. While compute itself is cheap compared to that. With this instance size coverage runs within 1 hour.